apps/cowswap-frontend/src/modules/hooksStore/validateHookDappManifest.tsx

```diff
@@ -21,34 +24,30 @@ export function validateHookDappManifest(
     const emptyFields = MANDATORY_DAPP_FIELDS.filter((field) => typeof dapp[field] === 'undefined')
 
     if (emptyFields.length > 0) {
-      return `${emptyFields.join(',')} fields are no set.`
+      return ERROR_MESSAGES.MISSING_REQUIRED_FIELDS(emptyFields)
     } else {
       if (
         isSmartContractWallet === true &&
         typeof conditions.walletCompatibility !== 'undefined' &&
         !conditions.walletCompatibility.includes(HookDappWalletCompatibility.SMART_CONTRACT)
       ) {
-        return 'The app does not support smart-contract wallets.'
+        return ERROR_MESSAGES.SMART_CONTRACT_INCOMPATIBLE
       } else if (!isHex(dapp.id) || dapp.id.length !== HOOK_DAPP_ID_LENGTH) {
-        return <p>Hook dapp id must be a hex with length 64.</p>
+        return ERROR_MESSAGES.INVALID_HOOK_ID
       } else if (chainId && conditions.supportedNetworks && !conditions.supportedNetworks.includes(chainId)) {
-        return <p>This app/hook doesn't support current network (chainId={chainId}).</p>
-      } else if (conditions.position === 'post' && isPreHook === true) {
-        return (
-          <p>
-            This app/hook can only be used as a <strong>post-hook</strong> and cannot be added as a pre-hook.
-          </p>
+        return ERROR_MESSAGES.NETWORK_COMPATIBILITY_ERROR(
+          chainId,
+          getChainInfo(chainId).label,
+          conditions.supportedNetworks.map((id) => ({ id, label: getChainInfo(id).label })),
         )
+      } else if (conditions.position === 'post' && isPreHook === true) {
+        return ERROR_MESSAGES.HOOK_POSITION_MISMATCH('post')
       } else if (conditions.position === 'pre' && isPreHook === false) {
-        return (
-          <p>
-            This app/hook can only be used as a <strong>pre-hook</strong> and cannot be added as a post-hook.
-          </p>
-        )
+        return ERROR_MESSAGES.HOOK_POSITION_MISMATCH('pre')
       }
     }
   } else {
-    return 'Manifest does not contain "cow_hook_dapp" property.'
+    return ERROR_MESSAGES.INVALID_MANIFEST
```

Comment:
nit: normally is nicer to assert first the negative cases so the early return are close to their check. What I mean here is that is a bit hard to know which case is this else is coming from cause this function has a lot of nested IF/ELSE statements

An alternative is to assert first like:

```
if (!dapp) {
  return ERROR_MESSAGES.INVALID_MANIFEST
}

//now we can continue, with the case out of the way and we don't need to nest so many IF/ELSE statements
```

Following this logic, many of the other checks can also be simplified I believe

Reply:
Refactored this now.